### PR TITLE
C2: Zod validation foundation — parseCapabilityInput + schema parity

### PR DESCRIPTION
**Pull request by totallynotjon**
## Summary

Migrates all 16 read-capability `run()` functions from ad-hoc `requireString`/`asString`/`asPositiveInt` helpers to a unified `parseCapabilityInput(schema, input)` pattern backed by Zod schemas. The advertised `inputSchema` is now derived from the same Zod schema via `toInputSchema`, so advertisement and enforcement are always in sync.

## Changes

- **`inputHelpers.ts`**: adds `parseCapabilityInput`, `toInputSchema`, `requiredStringField`, `optionalStringField`, `optionalClampedInt`, `ORG_ID_FIELD` exports; drops `.int()` from `optionalClampedInt` to keep JSON Schema type as `"number"` (matching legacy)
- **`rewstReadCapabilities.ts`**: all 16 capabilities migrated; custom pre-parse guards for `modelType` (friendly enum errors) and `createdSince` (epoch-ms rejection)
- **`rewstReadCapabilities.schemaParity.test.ts`**: new frozen-fixture parity test (16 tools × 4 checks = 64 assertions) — switched from `tdd.ts` to Mocha directly so it runs on the electron runner
- **`rewstReadCapabilities.test.ts`**: new capability-level tests covering happy paths, error messages, and edge cases
- **`openspec/specs/mcp-bridge/spec.md`**: updated "Validate tool inputs defensively" requirement to reflect schema-driven approach; added invalid-enum and missing-field scenarios
- **`CLAUDE.md`**: updated capability authoring guidance to document `parseCapabilityInput` pattern
- **`changelog.d/c2-zod-validation-foundation.md`**: changelog entry

## Test results

- vitest: 175 passing (9 files)
- electron unit grep `Unit: rewstReadCapabilities`: 90 passing
- type-check: clean
- changelog: valid

## Summary by CodeRabbit

* **New Features**
  * Tool input validation is now more consistent across read-only tools, with clearer, human-readable error messages when arguments are missing or invalid.
  * Numeric inputs are now handled more predictably, including automatic clamping and safer defaults for out-of-range values.

* **Bug Fixes**
  * Fixed several cases where malformed inputs could produce confusing validation output.
  * Improved handling of string trimming, enum checks, dates, and query variables so tools behave more reliably.

### CODE REVIEW

File: changelog.d/c2-zod-validation-foundation.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+category: Fixed
+---
+
+- **Rewst Buddy tool validation** now returns clearer errors for malformed read-tool arguments and keeps advertised schemas aligned with runtime validation.
```
[the remark this answers is not in the document]
Comment (totallynotjon) [reply changelog.d/c2-zod-validation-foundation.md]: Trimmed the changelog note to the user-facing behavior: clearer errors for malformed Rewst Buddy read-tool arguments.

File: src/capabilities/inputHelpers.test.ts [hunk 2]
```diff
@@ -265,3 +271,165 @@ suite('Unit: inputHelpers — requireStringAllowEmpty', () => {
...
+			properties: Record<string, { type: string }>;
+			required?: string[];
+		};
+		assert.ok(Array.isArray(result.required), 'required should be an array');
+		assert.ok(result.required!.includes('requiredField'), 'requiredField must be in required');
+		assert.ok(!result.required!.includes('optionalField'), 'optionalField must not be in required');
+		assert.strictEqual(result.properties['requiredField'].type, 'string');
+		assert.strictEqual(result.properties['optionalField'].type, 'string');
+	});
+});
```
[the remark this answers is not in the document]
Comment (totallynotjon) [reply src/capabilities/inputHelpers.test.ts:454]: Added a toInputSchema test using requiredStringField, optionalStringField, and optionalClampedInt so the production helper composition is covered.
